cpp/src/Slice/Grammar.y

```diff
@@ -1674,41 +1689,33 @@ enumerator
 : ICE_IDENTIFIER
 {
     auto ident = dynamic_pointer_cast<StringTok>($1);
-    auto ens = make_shared<EnumeratorListTok>();
-    ContainerPtr cont = currentUnit->currentContainer();
-    EnumeratorPtr en = cont->createEnumerator(ident->v, nullopt);
-    if (en)
-    {
-        ens->v.push_front(en);
-    }
-    $$ = ens;
+    EnumPtr cont = dynamic_pointer_cast<Enum>(currentUnit->currentContainer());
+    $$ = cont->createEnumerator(ident->v, nullopt);
 }
 | ICE_IDENTIFIER '=' enumerator_initializer
 {
     auto ident = dynamic_pointer_cast<StringTok>($1);
-    auto ens = make_shared<EnumeratorListTok>();
-    ContainerPtr cont = currentUnit->currentContainer();
+    EnumPtr cont = dynamic_pointer_cast<Enum>(currentUnit->currentContainer());
```

Comment:
You don't check that cont is not null because an enumerator can only be defined in an Enum container?

Reply:
Yes, by the time we hit this parsing rule, if there isn't a container, something has already gone horribly wrong.